CMSIS/NN/Source/PoolingFunctions/arm_avgpool_s8.c

```diff
@@ -228,51 +258,97 @@ arm_status arm_avgpool_s8(const int dim_src_height,
                           int8_t *dst)
 {
 
-/* Reference C code adapted from CMSIS-NN arm_avepool_q7_HWC.
- */
-    (void)bufferA;
-    int16_t   i_ch_in, i_x, i_y;
-    int16_t   k_x, k_y;
+#if defined (ARM_MATH_DSP)
+
+  /* Run the following code for Cortex-M4 and Cortex-M7
+   */
+  int16_t k_x, k_y, i_x, i_y;
 
-    for (i_y = 0; i_y < dim_dst_height; i_y++)
+  for (i_y = 0; i_y < dim_dst_height; i_y++)
+  {
+    for (i_x = 0; i_x < dim_dst_width; i_x++)
     {
-        for (i_x = 0; i_x < dim_dst_width; i_x++)
+      int count = 0;
+
+      for (k_y = i_y * stride_height - padding_height; \
+           k_y < i_y * stride_height - padding_height + dim_kernel_height; k_y++)
+      {
+        for (k_x = i_x * stride_width - padding_width; \
+             k_x < i_x * stride_width - padding_width + dim_kernel_width; k_x++)
         {
-            for (i_ch_in = 0; i_ch_in < ch_src; i_ch_in++)
+          if (k_y >= 0 && k_x >= 0 && k_y < dim_src_height && k_x < dim_src_width)
```

Comment:
Lets move these check to outside the kernel loops..something similar to arm_max_pool_s8()

Reply:
Ok